integrations/extensions/docs/elasticsearch-install-and-setup/python-document-ingestion/README.md

```diff
@@ -0,0 +1,184 @@
+# Working with PDF and Office Documents in Elasticsearch
+
+## Table of contents:
+* [Step 1: Create a python virtual environment and upgrade pip](#step-1-create-a-python-virtual-environment-and-upgrade-pip)
+* [Step 2: Create all the necessary environment variables](#step-2-create-all-the-necessary-environment-variables)
+* [Step 3: Create an ELSER ingest pipeline with an inference processor](#step-3-create-an-elser-ingest-pipeline-with-an-inference-processor)
+* [Step 4: Create an index for storing tokens as rank features](#step-4-create-an-index-for-storing-tokens-as-rank-features)
+* [Step 5: Running the python notebook](#step-5-running-the-python-notebook)
+
+## Pre-requisites:
+
+1. The following tutorial assumes there exists a folder of documents that you would like to index into Elasticsearch. These can be PDF, Microsoft Office, OpenOffice, HTML, Text files etc... See full list of supported types [here](https://tika.apache.org/2.9.1/formats.html) 
+
+	Optional: If using IBM COS or other Cloud Object Storage for your files, you can follow instructions to use [s3fs](https://cloud.ibm.com/docs/cloud-object-storage?topic=cloud-object-storage-s3fs) or [rclone](https://cloud.ibm.com/docs/cloud-object-storage?topic=cloud-object-storage-rclone) to synchronize your data to a local mounted filesystem
+
+2. You will need a working install of [python](https://www.python.org/downloads/) and Java 7+ on your machine.
+
+## Introduction
+
+Before you start, ensure you have set up your Elasticsearch cluster:
+
+* For Elasticsearch on IBM Cloud, please refer to [ICD-elasticsearch-install-and-setup](./ICD_Elasticsearch_install_and_setup.md) for more details.
+* For Elasticsearch (watsonx Discovery) on CloudPak, please refer to [watsonx-discovery-install-and-setup](./watsonx_discovery_install_and_setup.md) for more details.
+
+
+### Step 1: Create a python virtual environment and upgrade pip
+
+`venv` allows you to manage separate package installations for different projects. It creates a “virtual” isolated Python installation. When you switch projects, you can create a new virtual environment which is isolated from other virtual environments. You benefit from the virtual environment since packages can be installed confidently and will not interfere with another project’s environment.
+
+```bash
+python3 -m venv .venv
+source .venv/bin/activate
+python3 -m pip install --upgrade pip
+```
+
+
+### Step 2: Create all the necessary environment variables
+
+The `ES_INDEX_NAME` and `ES_PIPELINE_NAME` variables can be whatever you would like to name your index and ingestion pipeline for use throughout this process/guide as references in various steps below.
+
+  ```bash
+  export ES_URL=https://<hostname:port>
+  export ES_USER=<username>
+  export ES_PASSWORD=<password>
+  export ES_CACERT=<path-to-your-cert>
+  export ES_INDEX_NAME=<name-of-index>
+  export ES_PIPELINE_NAME=<name-of-ingest-pipeline>
```

Comment:
The notebook also requires a `DOCS_DIR` environment variable pointing to the documents.